apps/sms/services/js/moz_mobile_message/messaging_service.js

```diff
@@ -0,0 +1,191 @@
+/*global BridgeServiceMixin, Settings */
+'use strict';
+
+if (!('BridgeServiceMixin' in self)) {
+  importScripts('/services/js/bridge_service_mixin.js');
+}
+
+(function(exports) {
+
+const SERVICE_NAME = 'messaging-service';
+const MOBILE_MESSAGE_CLIENT_NAME = 'mozMobileMessageShim';
+
+const METHODS = Object.freeze([
+  'sendSMS', 'sendMMS', 'resendMessage'
+]);
+
+function ensureBridge() {
+  if (!('bridge' in self)) {
+    importScripts('/lib/bridge.js');
+  }
+}
+
+// 0 is a valid value so we need to take care to not consider it as a falsy
+// value. We want to return null for anything that's not a number or a
+// string containing a number.
+function sanitizeServiceId(serviceId) {
+  if (serviceId == null || // null or undefined
+      isNaN(+serviceId)) {
+    serviceId = null;
+  } else {
+    serviceId = +serviceId;
+  }
+
+  return serviceId;
+}
+
+function getSendOptionsFromServiceId(serviceId) {
+  var sendOpts;
+
+  if (serviceId != null && // not null, not undefined
+      Settings.hasSeveralSim()) {
+    sendOpts = { serviceId: serviceId };
+  }
+
+  return sendOpts;
+}
+
+var MessagingService = {
+  /**
+   * Initializes our service using the Service mixin's initService and
+   * requests mozMobileMessage shim as client.
+   */
+  init() {
+    ensureBridge();
+
+    this.initService();
+
+    this.mozMobileMessageClient = bridge.client(MOBILE_MESSAGE_CLIENT_NAME);
```

Comment:
> optional nit: since mozMobileMessageClient is private and we started to use Symbols, we can use smth like const priv = Object.freeze({ mozMobileMessageClient: Symbol("mozMobileMessageClient")}). But it's veeeery optional, so you can ignore if you want :)

Because the file will be in its separate window, I don't think we need to be extra-careful here. I'd say simpler is better in this case :)

I agree with you about the logic to find the endpoint; is there a plugin for this in threads.js or do we need to implement ourselves. For sure there was an implementation of this in the previous version of threads.js